CopilotKit/packages/react-textarea/src/components/base-copilot-textarea/base-copilot-textarea.tsx

```diff
@@ -128,8 +128,54 @@ const BaseCopilotTextareaWithHoveringContext = React.forwardRef(
       };
     }, [props.suggestionsStyle]);
 
+   const toggleFormat = (editor: Editor, format: string) => {
+  const isActive = isFormatActive(editor, format);
+  Transforms.setNodes(
+    editor,
+    { [format]: isActive ? null : true },
+    { match: Text.isText, split: true }
+  );
+};
+
+const isFormatActive = (editor: Editor, format: string) => {
+  const [match] = Editor.nodes(editor, {
+    match: (n) => n[format] === true,
+    universal: true,
+  });
+  return !!match;
+};
+
+const boldElement = (props: RenderElementProps) => {
+  return (
+    <strong
+      {...props.attributes}
+      onClick={() => toggleFormat(props.editor, 'bold')}
+      style={{ fontWeight: isFormatActive(props.editor, 'bold') ? 'bold' : 'normal' }}
+    >
+      {props.children}
+    </strong>
+  );
+};
+
+const italicElement = (props: RenderElementProps) => {
+  return (
+    <em
+      {...props.attributes}
+      onClick={() => toggleFormat(props.editor, 'italic')}
+      style={{ fontStyle: isFormatActive(props.editor, 'italic') ? 'italic' : 'normal' }}
+    >
+      {props.children}
+    </em>
+  );
+};
+
+
     const renderElementMemoized = useMemo(() => {
-      return makeRenderElementFunction(suggestionStyleAugmented);
+      return makeRenderElementFunction({
```

Comment:
looks like `makeRenderElementFunction` still only takes a single input (`suggestionStyle`).
So passing it `boldElement` and `italicElement` is incorrect.

We have 2 options here:
1. The pre-existing approach uses CSS styling. We can mirror what we have for the suggestion element with bold/italics 

2. I actually like the approach you started here - using <strong> and <em> to demarcate bold/italics, and carrying over the remaining stylings.

Assuming it works, there's no need to pass boldElement / italicsElement to the `makeRenderElementFunction` function.

Reply:
Ya sure ataibarkai I will first setup it locally and test it on my local and try to make changes what you have mentioned. If I need any help definitely I will ask you.

Thanks.